velox/functions/prestosql/tests/MapFromEntriesTest.cpp

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/functions/prestosql/tests/utils/FunctionBaseTest.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::test;
+using namespace facebook::velox::functions::test;
+
+namespace {
+class MapFromEntriesTest : public FunctionBaseTest {
+ protected:
+  void evaluateExprAndAssert(
+      const VectorPtr& expected,
+      const std::string& expression,
+      const std::vector<VectorPtr>& input) {
+    auto result = evaluate<MapVector>(expression, makeRowVector(input));
+    assertEqualVectors(expected, result);
+  }
+
+  // Evaluate an expression only, usually expect error thrown.
+  void evaluateExpr(
+      const std::string& expression,
+      const std::vector<VectorPtr>& input) {
+    evaluate(expression, makeRowVector(input));
+  }
+
+  void evaluateExprWithConstantArray(
+      const std::vector<MapVectorPtr>& expectedVectors,
+      const std::string& expression,
+      const VectorPtr& input,
+      const vector_size_t kConstantSize = 1'000) {
+    auto evaluateConstant = [&](vector_size_t row, const VectorPtr& vector) {
+      return evaluate(
+          expression,
+          makeRowVector(
+              {BaseVector::wrapInConstant(kConstantSize, row, vector)}));
+    };
+    for (auto row = 0; row < expectedVectors.size(); row++) {
+      auto result = evaluateConstant(row, input);
+      auto expectedConstant =
+          BaseVector::wrapInConstant(kConstantSize, 0, expectedVectors[row]);
+      assertEqualVectors(expectedConstant, result);
+    }
+  }
+};
+} // namespace
+
+TEST_F(MapFromEntriesTest, intKeyAndIntValue) {
+  auto rowType = ROW({INTEGER(), INTEGER()});
+  std::vector<std::vector<variant>> data = {
+      {variant::row({1, 11}), variant::row({2, 22}), variant::row({3, 33})}};
+  auto input = makeArrayOfRowVector(rowType, data);
+  auto expected =
+      makeMapVector<int32_t, int32_t>({{{1, 11}, {2, 22}, {3, 33}}});
+  evaluateExprAndAssert(expected, "map_from_entries(C0)", {input});
+}
+
+TEST_F(MapFromEntriesTest, intKeyAndVarcharValue) {
```

Comment:
I'm seeing multiple tests cases that differ only in the types of keys and values. Since the implementation of the function is agnostic to these, I think one test case is sufficient.